orangecontrib/timeseries/widgets/owspiralogram.py

```diff
@@ -426,6 +428,22 @@ def commit(self):
         self.Outputs.time_series.send(
             self.data[self.indices] if self.data else None)
 
+    @classmethod
+    def migrate_context(cls, context, version):
+        if version < 2:
+            values = context.values
+            context.values["agg_attr"] = values["agg_attr"][0][0]
+            _, type = values["agg_attr"]
+            ind, pos = values["agg_func"]
+            if type == 101: # discrete variable is always Mode in old settings
+                context.values["agg_func"] = ('Mode', pos)
+            elif type == 104: # time variable
+                time_map = {0: 'Mean', 2: 'Max', 3: 'Min', 4: 'Median',
+                            5: 'Mode', 13: 'Count nonzero', 14: 'Count defined'}
+                context.values["agg_func"] = (time_map.get(ind, 'Mean'), pos)
+            else:
+                context.values["agg_func"] = (list(AGG_OPTIONS)[ind], pos)
```

Comment:
Is the purpose of the code within `elif type == 104` to prevent `agg_func` to contain an unsupported aggregation? If so, the following would be more explicit.

```python
else:
    if type == 104 and ind not in {0, 2, 3, 4, 5, 13, 14}:  # time variable
        ind = 0
    context.values["agg_func"] =  (list(AGG_OPTIONS)[ind], pos)
```

On the other hand, it might not be needed at all because line 403 has the same effect.

Say "no" if you'd like to keep this code as it is.

Reply:
Yes, that is more explicit. I'll change it. Even though then I have to get rid of .get, which is the function of the month. 😉

Reply:
Again, yes. Line 403 was changed later as an additional inspiration (even though it is super necessary). I'll shorten it.